### DISCUSSION TITLE
Using SSH keys on windows

### DISCUSSION THREAD

**Discussion by kenha37** [copier-org/copier#887] [Q&A]
Hello, I have been debugging an ssh-key error when using Copier on Windows (both PowerShell and CMD).

The only way I can get the system to use SSH keys are by having SSH keys without passphrases.

Is this a general issue or have others found a solution for the problem?
And yes, I have checked that the keys where working when used in the CLI with git.

Error Message:
Command line: | 'C:\Program Files\Git\cmd\git.exe' clone --no-checkout ssh://GIT/algo/copier-python-docker.git 'C:\Users\$USER\AppData\Local\Temp\copier.vcs.clone.4vzpa4in' --filter=blob:none
Stderr:       | Cloning into 'C:\Users\$USER\AppData\Local\Temp\copier.vcs.clone.4vzpa4in'...
              | git@dcgit.dac.local: Permission denied (publickey).
              | fatal: Could not read from remote repository.
              |
              | Please make sure you have the correct access rights
              | and the repository exists.

**Comment by pawamoy** [comment]
Is there an SSH agent you could use on Windows? You could store the key's passphrase in there (temporarily of course), maybe that would help.

**Reply by kenha37** [reply]
I have no experience using an ssh-agent as I never have had problems storing the ssh keys in the User .ssh-folder.
I would prefer not having to add an ssh-agent to the mix to solve this problem permanently.

**Comment by pawamoy** [comment]
Does it fail immediately or after some time? Maybe you can feed your passphrase to the standard input of Copier :confused:?

**Reply by kenha37** [reply]
It seems to fail instantaneously (<1 sec) and I am not sure how you would add passphrase to the copier CLI, there doesn't seem to be any support for such actions in the documentation or the code it self.

Do you know if that is actually possible?

**Comment by yajo** [comment]
I've been using ssh agents for so long that I forgot you can actually be writing a password each time to unlock a key for ssh. So probably your user case isn't tested. Git is invoked as a subprocess quite often in copier, but maybe not in an interactive way that allows you to answer its prompts. 

That would count as a bug in copier. However your experience is always gonna be really better if you use an ssh agent. That makes me wonder if it's worth the time fixing it. 🤔

**Reply by kenha37** [reply]
I seems we are getting some contradiction experiences with copier and ssh-keys in our team, so I'll be moving to using an ssh-agent to get around the problem.

Thank you for following up on the question :)
